Problem validating JSON document against schema

~~~
2018-05-07T15:44:25.647+0200    ERROR    [request]    beater/handlers.go:384    error handling request    {"request_id": "f0472126-608a-41b8-81fb-1741d457e632", "error": "Problem validating JSON document against schema: I[#] S[#] doesn't validate with \"transaction#\"\n  I[#/transactions/0/spans/1/context/db/statement] S[#/properties/transactions/items/properties/spans/items/properties/context/properties/db/properties/statement/type] expected string or null, but got object"}
~~~

How I should proceed with debugging to find out what the issue is?
What information should I attach?

I'm receiving this errors on  1.0.0.beta1 gem.

This is a Rails app using Active Record with Postgresql

From the app side the errors look like this:

`apm-server's response: {"error":"Problem validating JSON document against schema: I[#] S[#] doesn't validate with \"transaction#\"\n  I[#/transactions/0/spans/13/context/db/statement] S[#/properties/transactions/items/properties/spans/items/properties/context/properties/db/properties/statement/type] expected string or null, but got object"}`

From the APM server side:

`ERROR   [request]       beater/handlers.go:384  error handling request  {"request_id": "60e2fc39-c2f4-43e1-992b-e3fea1741122", "error": "Problem validating JSON document against schema: I[#] S[#] doesn't validate with \"transaction#\"\n  I[#/transactions/0/spans/13/context/db/statement] S[#/properties/transactions/items/properties/spans/items/properties/context/properties/db/properties/statement/type] expected string or null, but got object"}`

--- a/lib/elastic_apm/normalizers/active_record.rb
+++ b/lib/elastic_apm/normalizers/active_record.rb
@@ -20,7 +20,7 @@ module ElasticAPM
           return :skip if %w[SCHEMA CACHE].include?(payload[:name])
 
           name = summarize(payload[:sql]) || payload[:name]
-          context = Span::Context.new(statement: payload[:sql].to_s)
+          context = Span::Context.new(statement: payload[:sql], type: 'sql')
           [name, @type, context]
         end
 

--- a/lib/elastic_apm/span/context.rb
+++ b/lib/elastic_apm/span/context.rb
@@ -6,7 +6,7 @@ module ElasticAPM
     class Context
       include NaivelyHashable
 
-      def initialize(**args)
+      def initialize(args)
         args.each do |key, val|
           send(:"#{key}=", val)
         end

--- a/lib/elastic_apm/spies/elasticsearch.rb
+++ b/lib/elastic_apm/spies/elasticsearch.rb
@@ -8,13 +8,15 @@ module ElasticAPM
       NAME_FORMAT = '%s %s'.freeze
       TYPE = 'db.elasticsearch'.freeze
 
+      # rubocop:disable Metrics/MethodLength
       def install
         ::Elasticsearch::Transport::Client.class_eval do
           alias perform_request_without_apm perform_request
 
           def perform_request(method, path, *args, &block)
             name = format(NAME_FORMAT, method, path)
-            context = Span::Context.new(statement: args[0])
+            statement = args[0].is_a?(String) ? args[0] : args[0].to_json
+            context = Span::Context.new(statement: statement)
 
             ElasticAPM.span name, TYPE, context: context do
               perform_request_without_apm(method, path, *args, &block)
@@ -22,6 +24,7 @@ module ElasticAPM
           end
         end
       end
+      # rubocop:enable Metrics/MethodLength
     end
 
     register(

--- a/spec/elastic_apm/spies/elasticsearch_spec.rb
+++ b/spec/elastic_apm/spies/elasticsearch_spec.rb
@@ -17,7 +17,7 @@ module ElasticAPM
       span, net_span = transaction.spans
 
       expect(span.name).to eq 'GET _search'
-      expect(span.context.statement).to eq(q: 'test')
+      expect(span.context.statement).to eq('{"q":"test"}')
 
       expect(net_span.name).to eq 'GET localhost'
 

